events/service/src/main/java/org/projectnessie/events/service/EventFactory.java

```diff
@@ -193,13 +199,24 @@ protected Event newContentRemovedEvent(
         .eventInitiator(extractName(user))
         .repositoryId(repositoryId)
         .properties(config.getStaticProperties())
-        .reference(ReferenceMapping.map(branch))
+        .reference(makeReference(branch))
         .hash(hash.asString())
         .contentKey(contentKey)
         .commitCreationTimestamp(commitTimestamp)
         .build();
   }
 
+  private static Reference makeReference(NamedRef refName) {
+    if (refName instanceof TagName) {
+      return Tag.of(refName.getName(), null);
```

Comment:
The `Branch` class from Nessie model is "annoying" because it conveys a hash; I chose to not include any hash though, because hashes are already set elsewhere and with events, we must be careful not to confuse hashes before the event with hashes after it. Putting hashes here would be ambiguous imho.

Reply:
I see the point, but IMHO having a `Tag` of `Branch` as a "result" object without a hash on the client side is strange. 

Could we instead add tests that the hashes in different attributes agree?

Reply:
We could, but there will be some hashes missing: we don't have the hashes from the source ref for merges and transplants, simply because the results from the version store do not include these. We'd need a small change to the version store API, I can look into it in a separate PR.

Reply:
FYI #9592

Reply:
Updated.